test/common/http/header_map_impl_test.cc

```diff
@@ -1118,51 +1212,97 @@ TEST(HeaderMapImplTest, TestInlineHeaderAdd) {
   EXPECT_TRUE(foo.Path() != nullptr);
 }
 
+TEST(HeaderMapImplTest, ClearHeaderMap) {
```

Comment:
Can you beef this up a bit more to cover more cases including multiple headers, a mix of inline and random, etc.?